activerecord/lib/active_record/connection_adapters/abstract/connection_pool.rb

```diff
@@ -31,12 +31,22 @@ def lazily_set_schema_cache
     class NullPool # :nodoc:
       include ConnectionAdapters::AbstractPool
 
+      class NullConfig # :nodoc:
+        def method_missing(*)
+          nil
+        end
+      end
+      NULL_CONFIG = NullConfig.new # :nodoc:
+
       attr_accessor :schema_cache
 
       def connection_class; end
       def checkin(_); end
       def remove(_); end
       def async_executor; end
+      def db_config
```

Comment:
This change sucks. It is only required because when checkind out a connection form the pool, we first create the object, check if the database version satisfies the requirements, and then we assign the pool to the connection.

When we are checking the database version, we are doing a query, so we try to annotate the query with the tags, but the pool isn't assigned yet, so `pool.db_config` would return an exception.

This is because the NullPool doesn't satisfy the ConnectionPool interface, which we should fix.

As the only queries that happen without a pool are very likely internal queries I'm retuning an object that respond to anything, without caring if that object have the same interface of the real configs.